google-cloud-firestore/src/main/java/com/google/cloud/firestore/spi/v1/GrpcFirestoreRpc.java

```diff
@@ -126,18 +122,7 @@ public GrpcFirestoreRpc(final FirestoreOptions options) throws IOException {
 
         clientContext = ClientContext.create(settingsBuilder.build());
       }
-      ApiFunction<UnaryCallSettings.Builder<?, ?>, Void> retrySettingsSetter =
-          new ApiFunction<Builder<?, ?>, Void>() {
-            @Override
-            public Void apply(UnaryCallSettings.Builder<?, ?> builder) {
-              builder.setRetrySettings(options.getRetrySettings());
-              return null;
-            }
-          };
-      FirestoreStubSettings.Builder firestoreBuilder =
-          FirestoreStubSettings.newBuilder(clientContext)
-              .applyToAllUnaryMethods(retrySettingsSetter);
-      firestoreStub = GrpcFirestoreStub.create(firestoreBuilder.build());
+      firestoreStub = GrpcFirestoreStub.create(clientContext);
```

Comment:
@BenWhitehead This might actually be enough to solve the underlying issue in `close()`, which means that we might not need to expose a separate shutdown API. With the original code, the newly created `firestoreStub` does not reference the existing background resources of the `clientContext` and thus the shutdown call does not actually do anything.

I also do not know what the consequence of not providing the retry settings builder is.

Reply:
I was able to add this back by adding the existing logic from `close` that manually closed the background resources.